### PR TITLE
Use netexec container in http lifecycle hook test.

### LINKED ISSUE #33636: [k8s.io] Container Lifecycle Hook when create a pod with lifecycle hook when it is http hook should execute poststart http hook properly {E2eNode Suite}
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/kubelet-gce-e2e-ci/9733/

Failed: [k8s.io] Container Lifecycle Hook when create a pod with lifecycle hook when it is http hook should execute poststart http hook properly {E2eNode Suite}

```
/var/lib/jenkins/workspace/kubelet-gce-e2e-ci/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e_node/lifecycle_hook_test.go:134
Expected error:
    <*errors.errorString | 0xc820414690>: {
        s: "pod ran to completion",
    }
    pod ran to completion
not to have occurred
/var/lib/jenkins/workspace/kubelet-gce-e2e-ci/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/framework/pods.go:66
```

### PR DESCRIPTION
**Pull request by Random-Liu**
Fixes https://github.com/kubernetes/kubernetes/issues/33636.

The original test is using `"echo -e \"HTTP/1.1 200 OK\n\" | nc -l -p 1234` as a simple http server.

However, it seems that this is not very reliable, which may response before golang thinks it should.
So we get the error:
```
I1106 06:14:13.325397    2096 logs.go:41] Unsolicited response received on idle HTTP channel starting with "HTTP/1.1 200 OK\n\n"; err=<nil>
```

This PR changes the test to use the `netexec` container which is a simple http server written by golang and used in many of our networking e2e test. It should be more reliable.
Mark 1.5 since this is fixing a 1.5 release blocking issue. Mark P0 to match the original issue.

@dchen1107

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/37109)

### CODE REVIEW

**Comment by Random-Liu** [comment]
@yujuhong Could you help me review this, since you've reviewed the initial PR adding this test. :)

**Comment by yujuhong** [comment]
Will review after lunch.

**Comment by dims** [comment]
@k8s-bot gci gce e2e test this

**Comment by Random-Liu** [comment]
@yujuhong Please take a look. :)

File: test/e2e_node/lifecycle_hook_test.go [hunk 1]
```diff
@@ -94,64 +99,112 @@ var _ = framework.KubeDescribe("Container Lifecycle Hook", func() {
...
 				podHandleHookRequest = podClient.CreateSync(podHandleHookRequest)
 				targetIP = podHandleHookRequest.Status.PodIP
 			})
+			testPodWithHttpHook := func(podWithHook *api.Pod) {
+				By("create the pod with lifecycle hook")
+				podClient.CreateSync(podWithHook)
+				if podWithHook.Spec.Containers[0].Lifecycle.PostStart != nil {
+					By("check poststart hook")
+					Eventually(func() error {
+						return podClient.MatchContainerOutput(podHandleHookRequest.Name, podHandleHookRequest.Name,
```
Comment (yujuhong) [inline test/e2e_node/lifecycle_hook_test.go]: I know they are the same, but the second argument should really be the container name.

File: test/e2e_node/lifecycle_hook_test.go [hunk 2]
```diff
@@ -94,64 +99,112 @@ var _ = framework.KubeDescribe("Container Lifecycle Hook", func() {
...
 			var targetIP string
+			var podHandleHookRequest *api.Pod
 			BeforeEach(func() {
-				By("cleanup the container to handle the HTTPGet hook request.")
-				podHandleHookRequest := getLifecycleHookTestPod("pod-handle-http-request",
-					[]string{"sh", "-c",
-						// Create test file when receive request on 1234.
-						fmt.Sprintf("echo -e \"HTTP/1.1 200 OK\n\" | nc -l -p 1234; touch %s", file),
+				By("create the container to handle the HTTPGet hook request.")
+				podHandleHookRequest = &api.Pod{
```
Comment (yujuhong) [inline test/e2e_node/lifecycle_hook_test.go]: Why defining the pod in `BeforeEach` when it doesn't rely on any variable?

File: test/e2e_node/lifecycle_hook_test.go [hunk 3]
```diff
@@ -94,64 +99,112 @@ var _ = framework.KubeDescribe("Container Lifecycle Hook", func() {
...
+						return podClient.MatchContainerOutput(podHandleHookRequest.Name, podHandleHookRequest.Name,
+							`GET /echo\?msg=poststart`)
+					}, podWaitTimeout, podCheckInterval).Should(BeNil())
+				}
+				By("delete the pod with lifecycle hook")
+				podClient.DeleteSync(podWithHook.Name, api.NewDeleteOptions(15), podWaitTimeout)
+				if podWithHook.Spec.Containers[0].Lifecycle.PreStop != nil {
+					By("check prestop hook")
+					Eventually(func() error {
+						return podClient.MatchContainerOutput(podHandleHookRequest.Name, podHandleHookRequest.Name,
```
Comment (yujuhong) [inline test/e2e_node/lifecycle_hook_test.go]: same here: the second argument should really be the container name.

File: test/e2e_node/lifecycle_hook_test.go [hunk 4]
```diff
@@ -94,64 +99,112 @@ var _ = framework.KubeDescribe("Container Lifecycle Hook", func() {
...
+							`GET /echo\?msg=prestop`)
+					}, podWaitTimeout, podCheckInterval).Should(BeNil())
```
Comment (yujuhong) [inline test/e2e_node/lifecycle_hook_test.go]: Does `DeleteSync` block till the pod has been deleted? If yes, checking the prestop hook should not need such a high timeout.

File: test/e2e_node/lifecycle_hook_test.go (hunk 2 above)
Comment (Random-Liu) [inline test/e2e_node/lifecycle_hook_test.go]: Will do.

File: test/e2e_node/lifecycle_hook_test.go (hunk 4 above)
Comment (Random-Liu) [inline test/e2e_node/lifecycle_hook_test.go]: Will do.

**Comment by Random-Liu** [comment]
@yujuhong Addressed comments.

**Comment by yujuhong** [comment]
LGTM. Squash?

**Comment by Random-Liu** [comment]
> LGTM. Squash?

ACK.

**Comment by calebamiles** [comment]
@saad-ali, we should probably get this in ASAP since it addresses https://github.com/kubernetes/kubernetes/issues/33636 which is a release blocker. Who do we need to get an `lgtm` on this?

**cc: @dims, @dchen1107 **

**Comment by saad-ali** [comment]
It is in the submit queue

**Comment by dims** [comment]
Thanks @yujuhong

**Pull request merged**